sdv/_utils.py

```diff
@@ -200,3 +201,80 @@ def _format_invalid_values_string(invalid_values, num_values):
             return f'{invalid_values[:num_values] + extra_missing_values}'
 
     return f'{invalid_values}'
+
+
+def _find_root_tables(relationships):
+    parent_tables = {rel['parent_table_name'] for rel in relationships}
+    child_tables = {rel['child_table_name'] for rel in relationships}
+    return parent_tables - child_tables
+
+
+def _get_relationship_idx_for_child(relationships, child_table):
+    return [idx for idx, rel in enumerate(relationships) if rel['child_table_name'] == child_table]
```

Comment:
Do we actually need the index here or can we pass the relationship itself?

Reply:
Yes, done in [995a684](https://github.com/sdv-dev/SDV/pull/1800/commits/995a684c2e6d9edff4a6daf8e631bd0a89467f12)